client/src/components/applications/profil/components/MapContainer.js

```diff
@@ -0,0 +1,153 @@
+import React, {Component} from 'react'
+import {Map, Marker, GoogleApiWrapper} from 'google-maps-react'
+
+export class MapContainer extends Component {
+  mapStyles = [
+    {elementType: 'geometry.fill', stylers: [{color: '#f1f4f5'}]},
+    {elementType: 'geometry.stroke', stylers: [{color: '#cddae3'}]},
+    {elementType: 'labels.text.fill', stylers: [{color: '#666666'}]},
+    {elementType: 'labels.text.stroke', stylers: [{color: '#ffffff'}]},
+    {
+      featureType: 'administrative',
+      elementType: 'geometry.stroke',
+      stylers: [{color: '#333333'}],
+    },
+    {
+      featureType: 'landscape',
+      elementType: 'geometry.stroke',
+      stylers: [{color: '#859fb4'}],
+    },
+    {
+      featureType: 'landscape.natural',
+      elementType: 'geometry',
+      stylers: [{color: '#f1f4f5'}],
+    },
+    {
+      featureType: 'landscape.man_made',
+      elementType: 'geometry.fill',
+      stylers: [{color: '#dae3ea'}],
+    },
+    {
+      featureType: 'road',
+      elementType: 'geometry.fill',
+      stylers: [{color: '#ffffff'}],
+    },
+    {
+      featureType: 'road',
+      elementType: 'geometry.stroke',
+      stylers: [{color: '#a5baca'}],
+    },
+    {
+      featureType: 'road.local',
+      elementType: 'geometry.fill',
+      stylers: [{color: '#ffffff'}],
+    },
+    {
+      featureType: 'transit.line',
+      elementType: 'geometry',
+      stylers: [{color: '#c5d1da'}],
+    },
+    {
+      featureType: 'transit.station',
+      elementType: 'geometry',
+      stylers: [{color: '#e6ecf1'}],
+    },
+    {
+      featureType: 'water',
+      elementType: 'geometry.fill',
+      stylers: [{color: '#ebf8ff'}],
+    },
+    {
+      featureType: 'water',
+      elementType: 'labels.text.fill',
+      stylers: [{color: '#859fb4'}],
+    },
+    {
+      featureType: 'landscape.natural',
+      elementType: 'labels',
+      stylers: [{visibility: 'off'}],
+    },
+    {
+      featureType: 'road.highway',
+      elementType: 'labels',
+      stylers: [{visibility: 'off'}],
+    },
+    {
+      featureType: 'road.arterial',
+      elementType: 'labels',
+      stylers: [{visibility: 'off'}],
+    },
+  ]
+
+  static defaultProps = {
+    center: {lat: 48.6, lng: 19.5},
+    google: null,
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    if (nextProps.google !== this.props.google) {
+      return true
+    }
+
+    if (this.props.assets !== nextProps.assets) {
+      this.zoom = 7
```

Comment:
I don't like this optimization. You are trying to save time by not recomputing markers when assets don't change. But creating markers in not a slow operation (it's mapping through ~500 entities). Slow operation is marker rendering which requires clustering optimization. Best option is to remove shouldComponentUpdate and derive the markers on every render...

Reply:
I will keep this for now. MapContainer wil be changed in next pull request.